streams/src/main/java/org/apache/kafka/streams/processor/internals/StateUpdater.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.kafka.streams.processor.internals;
+
+import java.time.Duration;
+import java.util.List;
+
+public interface StateUpdater {
+
+    /**
+     * Adds a task (active or standby) to the state updater.
+     *
+     * The state of the task will be updated.
+     *
+     * @param task task
+     */
+    void add(final Task task);
+
+    /**
+     * Removes a task (active and standby) from the state updater.
+     *
+     * A task is removed from the state updater irrespective of whether its state is up-to-date or not.
+     *
+     * @param task tasks to remove
+     */
+    void remove(final Task task);
```

Comment:
Each time a task is suspended in state `RESTORING`, we need to remove it from the state updater.